api/lib/subscription/slashing.js

```diff
@@ -38,20 +63,69 @@ class SlashingMonitor {
     )
 
     // requires some more logic to not spam the notifications if a validator is down for 1000 blocks
-    // this.client.subscribe(
-    //   { query: 'liveness.missed_blocks >= 1' },
-    //   (response) => {
-    //     const missedBlocks = response.events['liveness.address'].map(
-    //       (address, index) => ({
-    //         operatorAddress: address,
-    //         missedBlocks: response.events['liveness.missed_blocks'][index],
-    //         height: response.height
-    //       })
-    //     )
-    //     console.log('Missed block', missedBlocks)
-    //   }
-    // )
+    this.client.subscribe(
+      { query: `tm.event='NewBlockHeader' AND liveness.missed_blocks >= 1` },
+      async (response) => {
+        try {
+          const missedBlocks = response.events['liveness.address'].map(
+            (address, index) => ({
+              networkId: this.networkId,
+              operatorAddress: address,
+              blocks: response.events['liveness.missed_blocks'][index],
+              reason: 'missed_blocks',
+              height: response.data.value.header.height
+            })
+          )
+          const rows = await Promise.all(
+            missedBlocks.map(async (missedBlockEvent) => {
+              const lastMissedBlockEvent = await getLastMissedBlock(
+                this.networkId,
+                missedBlockEvent.operatorAddress
+              )
+              return aggregateMissedBlocks(
+                lastMissedBlockEvent,
+                missedBlockEvent
+              )
+            })
+          )
+          database(config)('').upsert('slashes', rows)
+          console.log('Wrote missed block events', rows.length)
```

Comment:
Mmm, I see. Isn't this confusing? We are handling missing blocks as "slashes", but they are two separate things. Should have also different tables, as I already commented

Reply:
they are not. you also get slashed for liveness. need to find the numbers

Reply:
But it could be a slash due to double signing. I think we should avoid mixing up the concepts

Reply:
oh, but that is why you are using this: `reason:{_eq:"missed_blocks"}`. OK, I see